mito-ai/src/Extensions/AiChat/ChatTaskpane.tsx

```diff
@@ -795,13 +823,27 @@ const ChatTaskpane: React.FC<IChatTaskpaneProps> = ({
     const markAgentForStopping = (): void => {
         // Signal that the agent should stop after current task
         shouldContinueAgentExecution.current = false;
+        
+        // IMMEDIATELY cancel all pending websocket requests
+        websocketClient.cancelPendingRequests();
+        
         // Update UI to show stopping state
         setAgentExecutionStatus('stopping');
     }
 
     const finalizeAgentStop = (): void => {
         // Notify user that agent has been stopped
         shouldContinueAgentExecution.current = false;
+        
+        // Ensure any remaining pending requests are cancelled
+        websocketClient.cancelPendingRequests();
+        
+        // Disconnect any active stream handlers
+        if (streamHandlerRef.current) {
+            websocketClient.stream.disconnect(streamHandlerRef.current, null);
+            streamHandlerRef.current = null;
+        }
```

Comment:
We currently don't do any streaming in agent mode, so we don't need this for now.